app/helpers/s3.py

```diff
@@ -0,0 +1,112 @@
+import boto3
+
+from app import app
+from config import MOBILIC_ENV
+
+SCW_REGION = app.config["SCW_REGION"]
+SCW_ENDPOINT = app.config["SCW_ENDPOINT"]
+SCW_ACCESS_KEY = app.config["SCW_ACCESS_KEY"]
+SCW_SECRET_KEY = app.config["SCW_SECRET_KEY"]
+
+# Each env (dev,staging,prod) has its S3 bucket
+BUCKET_NAME = f"mobilic-{MOBILIC_ENV}"
+
+PRESIGNED_URLS_EXPIRY_UPLOAD_S = 60
+PRESIGNED_URLS_EXPIRY_READ_S = 60
+
+
+S3 = boto3.client(
+    "s3",
+    endpoint_url=SCW_ENDPOINT,
+    aws_access_key_id=SCW_ACCESS_KEY,
+    aws_secret_access_key=SCW_SECRET_KEY,
+)
+
+
+class S3Client:
+    @staticmethod
+    def nb_pictures_for_control(control_id):
+        response = S3.list_objects_v2(
+            Bucket=BUCKET_NAME, Prefix=f"controls/control_{str(control_id)}/"
+        )
+
+        if "Contents" not in response:
+            return 0
+
+        # folder itself is counted as an item
+        return len(response["Contents"]) - 1
+
+    @staticmethod
+    def list_pictures_for_control(control_id, max_nb_pictures=None):
+        params = {
+            "Bucket": BUCKET_NAME,
+            "Prefix": f"controls/control_{str(control_id)}/",
+        }
+        if max_nb_pictures is not None:
+            params["MaxKeys"] = max_nb_pictures
+
+        response = S3.list_objects_v2(**params)
+
+        if "Contents" in response:
+            files = [item["Key"] for item in response["Contents"]]
+            return files
+        else:
+            return []
+
+    # generate presigned urls for the front to see picture
```

Comment:
suggestion(non-blocking): to avoid these comments, we can replace `generate_presigned_url_for_picture` by `generate_presigned_url_to_get_picture` and `generated_presigned_urls_for_control` by `generated_presigned_urls_to_upload_picture`